### PR TITLE
feature: 新增腾讯工蜂git.code.tencent.com的事件监听 #1180

### PR DESCRIPTION
**Pull request by mingshewhe**
#1180

### CODE REVIEW

File: src/backend/ci/core/common/common-pipeline/src/main/kotlin/com/tencent/devops/common/pipeline/pojo/element/trigger/CodeTGitWebHookTriggerElement.kt [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    @ApiModelProperty("excludeBranch", required = false)
+    val excludeBranchName: String?,
+    @ApiModelProperty("includePaths", required = false)
+    val includePaths: String?,
+    @ApiModelProperty("excludePaths", required = false)
+    val excludePaths: String?,
+    @ApiModelProperty("excludeUsers", required = false)
+    val excludeUsers: List<String>?,
+    @ApiModelProperty("eventType", required = false)
+    val eventType: CodeEventType?,
```
Comment (irwinsun) [inline src/backend/ci/core/common/common-pipeline/src/main/kotlin/com/tencent/devops/common/pipeline/pojo/element/trigger/CodeTGitWebHookTriggerElement.kt:70]: 这里最好不要用枚举 ，万一工蜂调整了事件，新增一些事件，可能会有序列化异常

File: src/backend/ci/core/repository/api-repository/src/main/kotlin/com/tencent/devops/repository/api/scm/ServiceGitResource.kt [hunk 2]
```diff
@@ -371,7 +374,10 @@ interface ServiceGitResource {
         tokenType: TokenTypeEnum,
         @ApiParam(value = "token", required = true)
         @QueryParam("token")
-        token: String
+        token: String,
+        @ApiParam(value = "仓库url", required = true)
```
Comment (irwinsun) [inline src/backend/ci/core/repository/api-repository/src/main/kotlin/com/tencent/devops/repository/api/scm/ServiceGitResource.kt:378]: 新旧接口的兼容考虑，参数不能是必填

**Comment by tencent-adm** [comment]
[CLA assistant check](https://cla.opensource.tencent.com/Tencent/bk-ci?pullRequest=2468) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.tencent.com/Tencent/bk-ci?pullRequest=2468) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.tencent.com/check/Tencent/bk-ci?pullRequest=2468) it.</sub>

File: src/backend/ci/core/repository/api-repository/src/main/kotlin/com/tencent/devops/repository/api/scm/ServiceGitResource.kt (hunk 2 above)
Comment (mingshewhe) [reply src/backend/ci/core/repository/api-repository/src/main/kotlin/com/tencent/devops/repository/api/scm/ServiceGitResource.kt:378]: 已修复

File: src/backend/ci/core/common/common-pipeline/src/main/kotlin/com/tencent/devops/common/pipeline/pojo/element/trigger/CodeTGitWebHookTriggerElement.kt (hunk 1 above)
Comment (mingshewhe) [reply src/backend/ci/core/common/common-pipeline/src/main/kotlin/com/tencent/devops/common/pipeline/pojo/element/trigger/CodeTGitWebHookTriggerElement.kt:70]: 插件的类型，只有几种类型，先用枚举

**Review by irwinsun — APPROVED** [review]

**Pull request merged**